docs/modules/ROOT/pages/references/architecture/sli_reporting.adoc

```diff
@@ -0,0 +1,322 @@
+A selector with no fields matches every cluster.
+A cluster is matched if for every key in the selector, the corresponding cluster fact matches the value specified in the selector.
+
+In addition to providing CRUD endpoints for maintaining the downtime windows, the service also provides an interface for serving all relevant downtime windows given just a cluster ID.
+This is achieved by querying the Lieutenant API in the background and retrieving the cluster facts from there, which can then be used to find matching downtime windows.
+
+If a new time window is created with the same non-null `ExternalID` as an existing time window, the existing window is updated instead.
+
+===== API endpoints
+
+The REST API endpoints of the Downtime Window API also correspond to the programmatic interface of this component within the SLO reporting service.
+
+====== `GET /downtime/`
+
+Lists all available downtime windows.
+
+Query parameters:
+
+* `from`: ISO timestamp defining the start time from which to include downtime windows.
+* `to`: ISO timestamp defining the end time until which to include downtime windows.
+
+The result contains all downtime windows that overlap with the time interval `[from, to]`.
+Notably, this also includes downtime windows that start before `from` but end after `from` (and conversely for `to`), as well as downtime windows that completely encompass the specified time interval.
+
+Returns:
+
+A list of downtime window records (as defined above).
+
+====== `GET /downtime/for-cluster/:cluster-id`
+
+Lists all available downtime windows that apply to a specific cluster (according to the match rules in `Affects`).
+
+Path parameters:
+* `cluster-id`: Lieutenant ID of the cluster for which to retrieve downtime windows.
+
+Query parameters:
+
+* `from`: ISO timestamp defining the start time from which to include downtime windows.
+* `to`: ISO timestamp defining the end time until which to include downtime windows.
+
+The result contains all downtime windows that overlap with the time interval `[from, to]`.
+Notably, this also includes downtime windows that start before `from` but end after `from` (and conversely for `to`), as well as downtime windows that completely encompass the specified time interval.
+
+Returns:
+
+A list of downtime window records (as defined above).
+
+====== `POST /downtime`
+
+Creates a new downtime window.
+
+Body parameters:
+
+The body of the request corresponds to the JSON example given above, minus the `ID` property.
+Any parameters marked as optional may be omitted.
+
+If an `ExternalID` is provided in the request body, and an existing downtime window shares the same `ExternalID`, then the existing window should be updated (replaced) instead of a new one created.
+Otherwise, a new record is created from the request body and assigned a new random ID.
+
+Returns:
+
+The newly created record, including ID.
+
+====== `POST /downtime/:id`
```

Comment:
If we want to keep `POST` as a proper POST, we should also support `PATCH` for partial updates of existing downtime records (e.g. just adjusting start or end time).